### PR TITLE
[new release] jose (0.7.0)

### PR DESCRIPTION
**Pull request by ulrikstrid**
JOSE implementation for OCaml and ReasonML

- Project page: <a href="https://ulrikstrid.github.io/reason-jose">https://ulrikstrid.github.io/reason-jose</a>
- Documentation: <a href="https://ulrikstrid.github.io/reason-jose">https://ulrikstrid.github.io/reason-jose</a>

##### CHANGES:

- Remove print statements that was used for debugging (by @phongphan)
- Make things safer by default, `of_string` will now return result, etc (by @anmonteiro)
- Fix deprecation warnings in libraries (by @anmonteiro)

### CODE REVIEW

File: packages/jose/jose.0.7.0/opam [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "base64" {>= "3.0.0"}
+  "dune" {>= "2.8" & >= "1.11"}
+  "eqaf" {>= "0.7"}
+  "mirage-crypto" {>= "0.10.0"}
+  "x509" {>= "0.13.0"}
+  "cstruct" {>= "4.0.0"}
+  "astring"
+  "yojson"
```
Comment (mseri) [inline packages/jose/jose.0.7.0/opam:20]:
```suggestion
  "yojson" {>= "1.6.0"}
```

Comment (mseri) [reply packages/jose/jose.0.7.0/opam:20]: Uses Yojson.Safe.t

**Comment by ulrikstrid** [comment]
I will return the changes upstream when we figure out the lower bounds, sorry about the churn. Is there a simple way I can test this in my own CI?

File: packages/jose/jose.0.7.0/opam [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+  "bisect_ppx" {with-test}
+  "alcotest" {with-test}
+  "junit" {with-test}
+  "junit_alcotest" {with-test}
+  "lwt" {with-test}
+  "odoc" {with-doc}
+]
+conflicts: [
+  "result" {< "1.5"}
+]
```
Comment (mseri) [inline packages/jose/jose.0.7.0/opam:32]:
```suggestion
  "rresult" {< "0.7.0"}
]
```

File: packages/jose/jose.0.7.0/opam [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+license: "MIT"
+homepage: "https://ulrikstrid.github.io/reason-jose"
+doc: "https://ulrikstrid.github.io/reason-jose"
+bug-reports: "https://github.com/ulrikstrid/reason-jose/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "base64" {>= "3.0.0"}
+  "dune" {>= "2.8" & >= "1.11"}
+  "eqaf" {>= "0.7"}
+  "mirage-crypto" {>= "0.10.0"}
```
Comment (mseri) [inline packages/jose/jose.0.7.0/opam:16]:
```suggestion
  "mirage-crypto" {>= "0.10.4"}
```
The current signature failure come from rresult used by ppx_sexp_conv.
We could try to add a confliuct with rresult < 0.7.0 or, if it is not a problem with you, by moving to mirage-crypto 0.10.4, you don't need ppx_sexp_conv any longer and it should solve the issue

Comment (ulrikstrid) [reply packages/jose/jose.0.7.0/opam:16]: That sounds reasonable to me, can't fix this from my phone but if you have time please make the change 😁

File: packages/jose/jose.0.7.0/opam [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+homepage: "https://ulrikstrid.github.io/reason-jose"
+doc: "https://ulrikstrid.github.io/reason-jose"
+bug-reports: "https://github.com/ulrikstrid/reason-jose/issues"
+depends: [
+  "ocaml" {>= "4.08.0"}
+  "base64" {>= "3.0.0"}
+  "dune" {>= "2.8" & >= "1.11"}
+  "eqaf" {>= "0.7"}
+  "mirage-crypto" {>= "0.10.4"}
+  "x509" {>= "0.13.0"}
```
Comment (mseri) [inline packages/jose/jose.0.7.0/opam:17]:
```suggestion
  "x509" {>= "0.15.0"}
```
Same rresult issue as the lther lower bound

**Comment by ulrikstrid** [comment]
The last failure just looks like a a flaky package?

I made a (apparently stupid) decision to prefix my internal utils with a `R` which is probably why rresult is broken, I'll change this until the next release.

**Comment by mseri** [comment]
Yes, now it works 🥳
No worries. Just keep the conflict there if you don’t want to rename the modules

**Comment by ulrikstrid** [comment]
Thank you @mseri ! I will return the fixes upstream for the future

**Pull request merged**